kms-provider-kroxylicious-inmemory/src/test/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/IntegrationTestingKmsServiceTest.java

```diff
@@ -0,0 +1,204 @@
+        var kms1 = service.buildKms(new IntegrationTestingKmsService.Config(kms1Id));
+        var kms2Id = UUID.randomUUID().toString();
+        var kms2 = IntegrationTestingKmsService.newInstance().buildKms(new IntegrationTestingKmsService.Config(kms2Id));
+        var key1 = kms1.generateKey();
+        assertNotNull(key1);
+        var key2 = kms2.generateKey();
+        assertNotNull(key2);
+
+        // when
+        CompletableFuture<InMemoryEdek> gen1 = kms1.generateDek(key2);
+        CompletableFuture<InMemoryEdek> gen2 = kms2.generateDek(key1);
+
+        // then
+
+        var e1 = assertThrows(ExecutionException.class, gen1::get,
+                "Expect kms to not generate dek for another kms's key");
+        assertInstanceOf(UnknownKeyException.class, e1.getCause());
+
+        var e2 = assertThrows(ExecutionException.class, gen2::get,
+                "Expect kms to not generate dek for another kms's key");
+        assertInstanceOf(UnknownKeyException.class, e2.getCause());
+
+        IntegrationTestingKmsService.delete(kms1Id);
+        IntegrationTestingKmsService.delete(kms2Id);
+    }
+
+    @Test
+    void shouldDecryptDeks() throws ExecutionException, InterruptedException {
+        // given
+        var kmsId = UUID.randomUUID().toString();
+        var kms = service.buildKms(new IntegrationTestingKmsService.Config(kmsId));
+        var kek = kms.generateKey();
+        assertNotNull(kek);
+        var pair = kms.generateDekPair(kek).get();
+        assertNotNull(pair);
+        assertNotNull(pair.edek());
+        assertNotNull(pair.dek());
+
+        // when
+        var decryptedDek = kms.decryptEdek(kek, pair.edek()).get();
+
+        // then
+        assertEquals(pair.dek(), decryptedDek, "Expect the decrypted DEK to equal the originally generated DEK");
+
+        IntegrationTestingKmsService.delete(kmsId);
+    }
+
+    @Test
+    void shouldSerializeAndDeserializeEdeks() throws ExecutionException, InterruptedException {
+        var kmsId = UUID.randomUUID().toString();
+        var kms = service.buildKms(new IntegrationTestingKmsService.Config(kmsId));
+        var kek = kms.generateKey();
+
+        var edek = kms.generateDek(kek).get();
+
+        var serde = kms.edekSerde();
+        var buffer = ByteBuffer.allocate(serde.sizeOf(edek));
+        serde.serialize(edek, buffer);
+        assertFalse(buffer.hasRemaining());
+        buffer.flip();
+
+        var deserialized = serde.deserialize(buffer);
+
+        assertEquals(edek, deserialized);
+
+        IntegrationTestingKmsService.delete(kmsId);
+    }
+
+    @Test
+    void shouldLookupByAlias() throws ExecutionException, InterruptedException {
+        var kmsId = UUID.randomUUID().toString();
+        var kms = service.buildKms(new IntegrationTestingKmsService.Config(kmsId));
+        var kek = kms.generateKey();
+
+        var lookup = kms.resolveAlias("bob");
+        var ee = assertThrows(ExecutionException.class, lookup::get);
```

Comment:
using `join` rather than `get` avoids the exceptions littering the test method signature.

```java
        assertThat(lookup)
                .failsWithin(Duration.ZERO)
                .withThrowableOfType(ExecutionException.class)
                .withCauseInstanceOf(UnknownAliasException.class)
                .withMessageContaining("bob");
```